src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1243,15 +1243,46 @@
         combined-list (vec (concat base-ext-list csr-ext-list))]
     (ensure-ext-list-has-cn-san subject combined-list)))
 
+(defn report-successful-cert-signing
+  "Log message and report to the activity service if available about successful cert signing."
+  [report-activity-or-nil message subject certname ip-address]
+  (let [commit {:service {:id "puppetserver"}
+                :subject {:id subject
+                          :name subject
+                          :type "users"}
+                :objects [{:type "node" :id certname :name certname}]
+                :events [{:type "sign-certificate"
+                          :what "node"
+                          :description (str "certificate_successfully_signed")
+                          :message message}]
+                :ip_address ip-address}]
+    (log/info message)
+    (if report-activity-or-nil
+      (report-activity-or-nil {:commit commit}))))
+
+(defn generate-cert-message-from-request
+  "Generate cert signing message from request parameters"
+  [request certname report-activity-or-nil]
+  (let [allow-unauthenticated (:allow-unauthenticated request)
+        auth-name (get-in request [:authorization :name])
+        rbac-subject (:rbac-subject request)
+        ip-address (:remote-addr request)
+        report-cert (partial report-successful-cert-signing report-activity-or-nil)]
+    (cond
+      (some? allow-unauthenticated) (report-cert (i18n/trs "Certificate autosigned for host {0}" certname) "CA" certname ip-address)
```

Comment:
I don't think `allow-unauthenticated` necessarily means it was autosigned (though it would be very dangerous if they disabled auth on the CA routes).